### ISSUE TITLE
Not enough colors to support more than 10 classes in classifier tree

### ISSUE THREAD

**Issue by jlubbersgeo** [parrt/dtreeviz#184]
As it stands it looks like there are only enough colors in ```colors.py``` to support up to 10 classes. My work has me trying to classify slightly more than that so it would be great if there was some support to visualize this. I think all that neeeds changing is the ```color_blind_friendly_colors```

To recreate the desired color format I've found that this does quite nicely and will support any current matplotlib colormap:
```python
import matplotlib
import numpy as np



def get_hex_colors(n_classes,cmap_name = "RdYlBu"):
    
    hex_colors = []
    for i in range(2, n_classes):
        cmap = matplotlib.cm.get_cmap(cmap_name, i)
        hex_colors.append(
            [
                matplotlib.colors.to_hex(rgb, keep_alpha=True)
                for rgb in cmap(np.arange(0, cmap.N))
            ]
        )

    hex_colors.insert(0, None)
    hex_colors.insert(0, None)
    
    return hex_colors
```

This then gets implemented in the ```trees.py``` as follows:
```python
n_classes = shadow_tree.nclasses()

colors['classes'] = get_hex_colors(n_classes+1,cmap) # added

color_values = colors['classes'][n_classes]
```
where ```cmap``` is just another argument in the function:
```python
viz = dtreeviz(
    rf.estimators_[tree_number],
    X,
    y,
    target_name="target_class",
    feature_names=features_list,
    class_names=class_names,
    cmap = 'viridis'

)
viz.view()
```

I don't have any strong opinions about the number of bins for histograms so I just manually adjusted this to be something like this:
```python
NUM_BINS = [0, 0, 10, 9, 8, 6, 6, 6, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5]
```
although there may be a more clever way to do this but if this list was just very long it would most likely be safe for the vast majority of users.  

I have it working on my local install of ```dtreeviz``` but was keen to see what the authors thought. Loving the package so far!

**Comment by tlapusan** [comment]
Hi @jlubbersgeo, thanks for your suggestion and effort.
Would be possible to create a PR? it would be easier to test for us also :)

**Comment by tlapusan** [comment]
thanks @jlubbersgeo, I will take a look on it in a few days.

**Comment by tlapusan** [comment]
implemented by this PR #185
